lib/AST/Decl.cpp

```diff
@@ -3586,72 +3584,57 @@ bool ClassDecl::inheritsSuperclassInitializers(LazyResolver *resolver) {
   return Bits.ClassDecl.InheritsSuperclassInits;
 }
 
-ObjCClassKind ClassDecl::checkObjCAncestry() const {
+AncestryOptions ClassDecl::checkAncestry() const {
   // See if we've already computed this.
-  if (Bits.ClassDecl.ObjCKind)
-    return ObjCClassKind(Bits.ClassDecl.ObjCKind - 1);
+  if (Bits.ClassDecl.Ancestry)
+    return AncestryOptions(Bits.ClassDecl.Ancestry - 1);
```

Comment:
This only works if you can guarantee that you'll never have all flags set. Can you (1) assert that below, and (2) include that in the doc comments of the mutually exclusive flags?

Reply:
Hmm, you're right that this doesn't make sense. I'll just use a separate bit for AncestryComputed

Reply:
Wow, I didn't even catch that.

Reply:
…and that's not even the right polarity for the necessary condition, so yeah, maybe let's not rely on this.